includes/RestApi/Themes/ThemeGeneratorController.php

```diff
@@ -45,17 +47,25 @@ public function register_routes() {
 	  */
 	public function generate_child_theme() {
 		// Ensure that we have sufficient data to generate a child theme.
-		$flow_data       = \get_option( 'nfd_module_onboarding_flow' )['data'];
-		$valid_flow_data = $this->validate_flow_data( $flow_data );
-		if ( ! $valid_flow_data ) {
+		$flow_data_option = \get_option( Options::get_option_name( 'flow' ), false );
+		if ( $flow_data_option === false || ! isset( $flow_data_option['data'] ) ) {
+			return new \WP_Error(
+				'nfd_onboarding_error',
+				'Flow data does not exist to generate a child theme.',
+				array( 'status' => 500 )
+			);
+		}
+
+		$flow_data = $this->validate_and_sanitize_flow_data( $flow_data_option['data'] );
```

Comment:
Had a doubt that as this function sets the $flow_data['theme']['template'] attribute should that be not saved in flow data?